filters/ratelimit/ratelimit.go

```diff
@@ -399,10 +399,8 @@ func (f *filter) Request(ctx filters.FilterContext) {
 	}
 
 	if !rateLimiter.AllowContext(ctx.Request().Context(), s) {
-		ctx.Serve(&http.Response{
-			StatusCode: http.StatusTooManyRequests,
-			Header:     ratelimit.Headers(&f.settings, rateLimiter.RetryAfter(s)),
-		})
+		ctx.Response().StatusCode = http.StatusTooManyRequests
```

Comment:
This does not seem right. The whole point of `ctx.Serve` is to shunt route and do not call subsequent filters and backend.